packages/xo-web/src/common/xo/index.js

```diff
@@ -821,13 +821,38 @@ export const setRemoteSyslogHost = (host, syslogDestination) =>
 export const setRemoteSyslogHosts = (hosts, syslogDestination) =>
   Promise.all(map(hosts, host => setRemoteSyslogHost(host, syslogDestination)))
 
-export const restartHost = (host, force = false, suspendResidentVms = false) =>
+export const restartHost = (host, force = false, suspendResidentVms = false, forceSmartReboot = false) =>
   confirm({
     title: _('restartHostModalTitle'),
     body: _('restartHostModalMessage'),
   }).then(
     () =>
-      _call('host.restart', { id: resolveId(host), force, suspendResidentVms })
+      _call('host.restart', { id: resolveId(host), force, suspendResidentVms, forceSmartReboot })
+        .catch(async error => {
+          if (
+            suspendResidentVms &&
+            !forceSmartReboot &&
+            operationBlocked.is(error, {
+              code: 'suspend',
```

Comment:
I don't think the error code would be `'suspend'`, do you confirm?

Reply:
Idk, looks like we're not using this error in the codebase.

Reply:
Yes, it's used here: https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-server/src/xo-mixins/api.mjs#L45
But apparently, the `error.code` that we receive from XAPI and that we pass there is just `'true'`, which isn't very useful.

Reply:
I think this is a misuse of this error and we should try to clean up things instead of making them messier.

Here's what we did for upgrading an XO Proxy: https://github.com/vatesfr/xen-orchestra/blob/f0f429a473c29aab4ea3b416012e923ace5c7667/packages/xo-server/src/xo-mixins/proxies.mjs#L203

What do you think of this solution?

I agree it's not ideal either :shrug: